src/Geometry/CSG/Intersection.jl

```diff
@@ -183,3 +183,18 @@ israyorigin(::RayOrigin) = true
 
 α(::RayOrigin{T}) where {T<:Real} = zero(T)
 Base.eltype(::RayOrigin{T}) where {T<:Real} = T
+
+
+"""
+Apply a Transform to an Intersection object
+"""
+function Base.:*(a::Transform{T}, intsct::Intersection{T,3})::Intersection{T,3} where {T<:Real}
+    u, v = uv(intsct)
+    i = interface(intsct)
+    if VERSION < v"1.6.0-DEV"
+        # TODO REMOVE
+        return @unionsplit OpticalInterface T i Intersection(α(intsct), a * point(intsct), Geometry.rotate(a, normal(intsct)), u, v, i, flippednormal = flippednormal(intsct))
```

Comment:
@BrianGun do you know what this is?

Reply:
In 1.5 if you had unions of concrete types with more than 4 (or maybe more than 3?) types then the compiler wouldn't optimize the code properly and would allocate on the heap. 

1.6 should allow unions of greater than 4 but I don't think we've tested this yet. If you get rid of it then performance on 1.5 for multithreaded code could be terrible, as in slower than single threaded. We have to decide how much we care about continuing to support 1.5 since 1.6 is so much better.